wasm/examples/centaur_game/index.ts

```diff
@@ -0,0 +1,354 @@
+import RiveCanvas from "../../../js/npm/canvas_advanced_single/canvas_advanced_single.mjs";
+import Centaur from "./centaur.riv";
+
+const appleRadius = 40;
+const appleRadiusSquared = appleRadius * appleRadius;
+
+async function main() {
+  let rive = await RiveCanvas();
+  // Instance the Rive runtime (this does WASM stuff).
+  // Get some rive file bytes (we uploaded piggy.riv into this
+  // sandbox, you can see it in the file list on the left.)
+  let fileBytes = new Uint8Array(
+    await (await fetch(new Request(Centaur))).arrayBuffer()
+  );
+  // Turn bytes into a runtime representation of the file.
+  const file = await rive.load(fileBytes);
+
+  // Get the character artboard.
+  const character = file.artboardByName("Character");
+  const arrow = file.artboardByName("Arrow");
+  const apple = file.artboardByName("Apple");
+  // Set frame origin to false so the arrow isn't offset by the
+  // origin in artboard space. This allows us to easily assume
+  // artboard space is worldspace.
+  arrow.frameOrigin = false;
+  const target = character.node("Look");
+  const arrows = new Set();
+  const apples = new Set();
+
+  const minApples = 1;
+  const maxApples = 5;
+  const appleBounds = apple.bounds;
+  // Spawn a random amount of apples (1-5) and creates that many
+  // new Apple state machine instances from "Apple" artboard
+  // Returns a set of custom objects for each apple instance with a random
+  // static position, the artboard instance of the apple,
+  // the specific instance of that state machine generated
+  // and the specific instance of the "explode" triggger input
+  function spawnApples() {
+    const count =
+      Math.round(Math.random() * (maxApples - minApples)) + minApples;
+    while (apples.size < count) {
+      const aplInstance = file.artboardByName("Apple");
+      const appleMachine = new rive.StateMachineInstance(
+        apple.stateMachineByName("Apple"),
+        aplInstance
+      );
+      let explodeTrigger;
+      for (let i = 0, l = appleMachine.inputCount(); i < l; i++) {
+        const input = appleMachine.input(i);
+        switch (input.name) {
+          case "Explode":
+            explodeTrigger = input.asTrigger();
+            break;
+
+          default:
+            break;
+        }
+      }
+
+      const appleInstance = {
+        x: -appleBounds.maxX + Math.random() * appleBounds.maxX * 3,
+        y: -appleBounds.maxY + Math.random() * -appleBounds.maxY,
+        artboard: aplInstance,
+        machine: appleMachine,
+        explodeTrigger: explodeTrigger,
+      };
+      appleMachine.advance(0);
+      appleInstance.artboard.advance(0);
```

Comment:
Can you not call advanceAndApply (just on the machine)?

Reply:
Oh - this must be new ish? It's not defined in bindings, but looking at the cpp def, yeah, we can.